common/src/main/java/com/wynntils/wynn/model/bossbar/BossBarModel.java

```diff
@@ -0,0 +1,219 @@
+                            max += 30;
+                        }
+                    }
+                }
+            };
+
+    public static final TrackedBar awakenedBar =
+            new TrackedBar(Pattern.compile("§fAwakening §7\\[§f(\\d+)/(\\d+)§7]"), TrackedBar.BarType.AWAKENED) {
+                @Override
+                public void onUpdateName(Matcher match) {
+                    try {
+                        current = Integer.parseInt(match.group(1));
+                        max = Integer.parseInt(match.group(2));
+                    } catch (NumberFormatException e) {
+                        WynntilsMod.error(String.format(
+                                "Failed to parse current and max for awakened bar %s (%s out of %s)",
+                                type, match.group(1), match.group(2)));
+                    }
+                }
+            };
+
+    public static final TrackedBar focusBar =
+            new TrackedBar(Pattern.compile("§eFocus §6\\[§e(\\d+)/(\\d+)§6]"), TrackedBar.BarType.FOCUS) {
+                @Override
+                public void onUpdateName(Matcher match) {
+                    try {
+                        current = Integer.parseInt(match.group(1));
+                        max = Integer.parseInt(match.group(2));
+                    } catch (NumberFormatException e) {
+                        WynntilsMod.error(String.format(
+                                "Failed to parse current and max for focus bar %s (%s out of %s)",
+                                type, match.group(1), match.group(2)));
+                    }
+                }
+            };
+
+    public static final TrackedBar corruptedBar =
+            new TrackedBar(Pattern.compile("§cCorrupted §4\\[§c(\\d+)%§4]"), TrackedBar.BarType.CORRUPTED) {
+                @Override
+                public void onUpdateName(Matcher match) {
+                    try {
+                        current = Integer.parseInt(match.group(1));
+                        max = 100;
+                    } catch (NumberFormatException e) {
+                        WynntilsMod.error(String.format(
+                                "Failed to parse current and max for corrupted bar %s (%s out of %s)",
+                                type, match.group(1), match.group(2)));
+                    }
+                }
+            };
+
+    private static final HashMap<UUID, TrackedBar> trackedBarsMap = new HashMap<>();
+
+    // FixPacketBugsFeature gets in the way if receiveCanceled is not set
+    @SubscribeEvent(receiveCanceled = true)
+    public static void onHealthBarEvent(BossHealthUpdateEvent event) {
+        ClientboundBossEventPacket packet = event.getPacket();
+
+        packet.dispatch(new TrackedBarHandler(event));
+    }
+
+    public record BarProgress(int current, int max, float progress) {}
+
+    private record TrackedBarHandler(BossHealthUpdateEvent event) implements ClientboundBossEventPacket.Handler {
+
+        public void add(
```

Comment:
Can you set `@Override` annotations on overridden methods? It helps seeing which methods come from the Handler.